src/main/kotlin/org/rust/lang/core/macros/MacroExpansionManager.kt

```diff
@@ -776,15 +802,34 @@ private class MacroExpansionServiceImplInner(
         return modData.toRsMod(project).singleOrNull()
     }
 
-    private fun RsModInfo.findMacroCall(macroIndex: MacroIndex, kind: RsProcMacroKind): RsPossibleMacroCall? {
+    private fun RsModInfo.findMacroCallByMacroIndex(macroIndex: MacroIndex, kind: RsProcMacroKind): RsPossibleMacroCall? {
         val modIndex = modData.macroIndex
         val ownerIndex = if (kind == DERIVE) macroIndex.parent else macroIndex
-        val parentIndex = ownerIndex.parent
-        val parent = if (MacroIndex.equals(parentIndex, modIndex)) {
-            modData.toRsMod(this).singleOrNull()
-        } else {
-            getExpansionFile(defMap, parentIndex)
+
+        val nestedIndices = mutableListOf<Int>()
+        val nearestKnownParent: RsMod = run {
+            var parentIndex = ownerIndex
+            while (true) {
+                parentIndex = parentIndex.parent
+                if (MacroIndex.equals(parentIndex, modIndex)) {
+                    return@run modData.toRsMod(this).singleOrNull()
+                }
+                if (parentIndex in defMap.macroCallToExpansionName) {
+                    return@run getExpansionFile(defMap, parentIndex)
+                }
+                nestedIndices += parentIndex.last
+            }
+            @Suppress("UNREACHABLE_CODE")
+            null
         } ?: return null
+
+        var parent: RsMod = nearestKnownParent
+        for (macroIndexInParent in nestedIndices) {
```

Comment:
Shouldn't we iterate `nestedIndices` in reverse order here? If yes lets add a test for it

Reply:
Adjusted. This case should now be covered by `test include file in included file in included file`